crates/chain/src/chain_data.rs

```diff
@@ -192,4 +205,88 @@ mod test {
             "confirmation_height is higher then it should be higher ord"
         );
     }
+
+    #[test]
+    fn test_sort_unconfirmed_chain_position() {
+        let mut v = vec![
+            ChainPosition::<ConfirmationBlockTime>::Unconfirmed {
+                first_seen: Some(5),
+                last_seen: Some(20),
+            },
+            ChainPosition::<ConfirmationBlockTime>::Unconfirmed {
+                first_seen: None,
+                last_seen: Some(30),
+            },
+            ChainPosition::<ConfirmationBlockTime>::Unconfirmed {
+                first_seen: Some(1),
+                last_seen: Some(10),
+            },
+            ChainPosition::<ConfirmationBlockTime>::Unconfirmed {
+                first_seen: Some(3),
+                last_seen: None,
+            },
+        ];
+
+        // Sort by first_seen: lowest first_seen first, None at the bottom
+        v.sort_by(|a, b| match (a, b) {
+            (
+                ChainPosition::<ConfirmationBlockTime>::Unconfirmed { first_seen: a, .. },
+                ChainPosition::<ConfirmationBlockTime>::Unconfirmed { first_seen: b, .. },
+            ) => a.unwrap_or(u64::MAX).cmp(&b.unwrap_or(u64::MAX)),
+            _ => Ordering::Equal,
+        });
```

Comment:
```suggestion
        v.sort();
```

This is enough. Not sure what the goal is of trying to Ord by different subsets of `ChainPosition`.

Reply:
Do we not need to be able to order by `first_seen` and `last_seen` for different use cases?

Reply:
@uvuvuwu The way the tests are implemented basically only tests the `Ord` impl of stdlib types are correct.

What we want to test is that the `ChainPosition` type has `Ord` implemented properly. `ChainPosition` is a user-facing struct, so we only want to order it chronologically (therefore by `first_seen`).

There may be other situations (such as during canonicalization or other niche use cases) where we MAY want to order by `last_seen`. This is out of the concern of `ChainPosition` and types can only have one `Ord` impl anyway.